### ISSUE TITLE
unexpectedly panick when compiling mysql_common

### ISSUE THREAD

**Issue by z0mbie42** [rust-lang/rust-clippy#4584]
Compiler panics when compiling `mysql_common 0.18.0`.

I tried to lint a project using `mysql_common` as dependencies using `cargo clippy` but I got the following error:
```
error: internal compiler error: src/librustc/ty/context.rs:516: node_type: no type for node `expr 5 (hir_id=HirId { owner: DefIndex(228), local_id: 255 })`

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:644:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
error: aborting due to previous error


note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.38.0-nightly (8a58268b5 2019-07-31) running on x86_64-apple-darwin

note: compiler flags: -C debuginfo=2 --crate-type lib

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `mysql_common`
```

I got the same error with the following Rust versions:
* `nightly-2019-08-01`
* `nightly-2019-08-17`
* `nightly-2019-09-11`
* `nightly-2019-09-13`

But it works with:

`nightly-2019-07-19`

**Comment by Manishearth** [comment]
Transferred to clippy

**Comment by Manishearth** [comment]
(It might be worth replacing the ICE message when compiling in clippy mode)

**Comment by mati865** [comment]
Tracked with details in https://github.com/rust-lang/rust-clippy/issues/4579

**Comment by z0mbie42** [comment]
@spease as as workaround you can do `cargo +nightly-2019-07-19 clippy` while waiting for a fix

**Comment by flip1995** [comment]
@Manishearth I have a fix for this ICE in #4590, currently waiting for a review 😉

**Comment by z0mbie42** [comment]
hi, @Manishearth, a rather noobish question but at which nightly release can we expect the fix since it's merged?

**Comment by flip1995** [comment]
#4590 is currently waiting on bors. After this a rust PR has to be created, which bumps the Clippy submodule in the compiler (Most of the time such PRs are only created if Clippy needs a fix because of a rustc breakage). Once this PR is merged, the fix will be in the next nightly.

TL;DR: Probably 2019-10-03 or 2019-10-04, no guarantees though.

**Comment by z0mbie42** [comment]
Ack, Thank you!
